src/astarte.rs

```diff
@@ -0,0 +1,439 @@
+// This file is part of Astarte.
+//
+// Copyright 2024 SECO Mind Srl
+//
+// SPDX-License-Identifier: Apache-2.0
+
+//! Astarte builder and configuration structures.
+
+use crate::cli::Config;
+use crate::math::BaseValue;
+use astarte_device_sdk::builder::{DeviceBuilder, DeviceSdkBuild};
+use astarte_device_sdk::store::SqliteStore;
+use astarte_device_sdk::transport::grpc::{Grpc, GrpcConfig};
+use astarte_device_sdk::transport::mqtt::{Credential, Mqtt, MqttConfig};
+use astarte_device_sdk::{Client, DeviceClient, DeviceConnection};
+use clap::ValueEnum;
+use color_eyre::eyre;
+use color_eyre::eyre::{ContextCompat, OptionExt, WrapErr};
+use serde::Deserialize;
+use std::path::{Path, PathBuf};
+use std::time::SystemTime;
+use std::{env, io};
+use tracing::{debug, error, info};
+
+/// Stream Rust test node identifier
+const STREAM_RUST_TEST_NODE_UUID: uuid::Uuid = uuid::uuid!("d72a6187-7cf1-44cc-87e8-e991936166dc");
```

Comment:
As a first implementation, we decided to embed the UUID, consequently having only once connection to the message hub. However, I think this can be changed (either in this PR or in a new one) if we want. What do you think @harlem88?

Reply:
Yep, you can change it in a new PR.